src/sensors/temperature/TMP36.cpp

```diff
@@ -0,0 +1,17 @@
+#include "TMP36.h"
+
+using namespace wlp;
+
+TMP36::TMP36(Board::AnalogPin analog_pin)
+    : pin(analog_pin) {}
+
+double TMP36::read_temperature() {
+    pin.powerup();
+    double reading = pin.sample();
+    double voltage = reading * 5.0;
+    voltage /= 1025.0;
+    float temperature_celsius = (voltage - 0.5) * 100.0;
+    pin.powerdown();
```

Comment:
it's definitely not a good idea to be powering up and down the pin each time it's called, because that would slow down reading values